build/builder.go

```diff
@@ -1011,15 +996,50 @@ func (b *Builder) buildShard(todo []*zoekt.Document, nextShardNum int) (*finishe
 
 	sortDocuments(todo)
 
-	for _, t := range todo {
+	for idx, t := range todo {
 		if err := shardBuilder.Add(*t); err != nil {
 			return nil, err
 		}
+
+		if idx%10_000 == 0 {
+			b.CheckMemoryUsage()
+		}
```

Comment:
Why do we check based on doc count and not, for example, time based? If it's time based we wouldn't have to check in two places?

Reply:
Good question! That approach could definitely work. What I liked about this: it gives fine-grained control over when we check memory. So we can check exactly when we are most concerned about an impending OOM, to maximize the chance we'll get useful data. In fact I just added another check after my latest round of research into memory usage (after loading the list of files to index).